test/TestProjects/CadlPetStore/Generated/PetsClient.cs

```diff
@@ -6,14 +6,15 @@
 #nullable disable
 
 using System;
+using System.Threading;
 using System.Threading.Tasks;
 using Azure;
 using Azure.Core;
 using Azure.Core.Pipeline;
 
 namespace CadlPetStore
 {
-    /// <summary> The Pets service client. </summary>
+    /// <summary> Manage your pets. You can delete or get the Pet from pet store. </summary>
     public partial class PetsClient
```

Comment:
Curious: where is the input specified?  
I am looking at https://github.com/microsoft/cadl/blob/main/packages/samples/petstore/petstore.cadl, and I'd like to understand better how you are choosing the name for the client, client options, and namespace.  Thanks!

Reply:
The input for the codegen is https://github.com/Azure/autorest.csharp/blob/66570505a5724751905560bfc8357fbf1834c81e/test/TestProjects/CadlPetStore/Generated/cadl.json.
Currently it does not respect https://github.com/microsoft/cadl/blob/main/packages/samples/petstore/petstore.cadl because the emitter is still using the stable version of compiler (but I think Crystal will update later).

Here is how we choosing:
Name for the client: https://github.com/Azure/autorest.csharp/blob/66570505a5724751905560bfc8357fbf1834c81e/test/TestProjects/CadlPetStore/Generated/cadl.json#L65, append `Client`;

Name for the client options: 
If there is only one client, append `ClientOptions` to client name; if there are more than one clients, use `InputNamespace.Name` https://github.com/Azure/autorest.csharp/blob/66570505a5724751905560bfc8357fbf1834c81e/test/TestProjects/CadlPetStore/Generated/cadl.json#L3

Name for namespace:
If there is `Configuration.Namespace` https://github.com/Azure/autorest.csharp/blob/66570505a5724751905560bfc8357fbf1834c81e/test/TestProjects/CadlPetStore/Generated/Configuration.json#L3, then use this, else use `InputNamespace.Name` https://github.com/Azure/autorest.csharp/blob/66570505a5724751905560bfc8357fbf1834c81e/test/TestProjects/CadlPetStore/Generated/cadl.json#L3